lib/mosaic-routing/src/main/java/com/csvreader/CsvReader.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2024 Fraunhofer FOKUS and others. All rights reserved.
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contact: mosaic@fokus.fraunhofer.de
+ */
+package com.csvreader;
+
+import com.opencsv.CSVParserBuilder;
+import com.opencsv.CSVReader;
+import com.opencsv.CSVReaderBuilder;
+import com.opencsv.exceptions.CsvValidationException;
+import com.opencsv.validators.LineValidator;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.Validate;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.Charset;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Replaces com.csvreader.CsvReader from javacsv with an implementation which uses CSVReader from opencsv.
+ * <br><br>
+ * By excluding javacsv jar and providing this class with the same full qualified name and same API as a supplement
+ * we are able to use opencsv instead without touching the code of graphhopper-reader-gtfs.
+ * Therefore, even though the IDE thinks this class is unused, it is actually used by graphhopper-reader-gtfs.
+ */
+public class CsvReader {
```

Comment:
Why do we want to replace `com.csvreader.CsvReader`? To get rid of a dependency?

This seems a little bit hacky 😅

Reply:
This comes from `net.sourceforge.javacsv/javacsv/2.0` package, which seems to be licensed under LGPL (which is not that kind of a deal actually), but I could not find its source code anymore which is requirement for us and Eclipse Foundation to check copyright.

Reply:
And yes it's hacky, but for future versions of graphhopper this should be eliminated by https://github.com/graphhopper/graphhopper/pull/3084